drivers/public_reports/app/models/public_reports/state_level_homelessness.rb

```diff
@@ -220,28 +336,36 @@ def parsed_pre_calculated_data
 
     def map_colors
       @map_colors ||= {}.tap do |m_colors|
-        slight = 0.0001
-        max_rate = parsed_pre_calculated_data['map_max_rate']
+        # slight = 0.000001
+        # ten_percent = 9.999999
+        # max_rate = parsed_pre_calculated_data.try(:[], 'map_max_rate') || map_max_rate
         # colors = chart_color_shades(:map_primary_color)
         colors = ['#FFFFFF']
         5.times do |i|
           colors << settings["color_#{i}"]
         end
-        m_colors[colors.first] = { description: '0%', range: (0..0) }
-        colors.drop(1).each.with_index do |color, i|
-          division_size = max_rate / colors.count
-          division_start = i * division_size
-          division_start = slight if division_start.zero?
-          division_end = (i + 1) * division_size
-          description = if i == colors.drop(2).count
-            "#{division_start.round}+"
-          elsif division_start == slight
-            "Any - #{division_end.round}%"
-          else
-            "#{division_start.round}% - #{division_end.round}%"
-          end
-          m_colors[color] = { description: description, range: (division_start..division_end) }
-        end
+
+        m_colors[colors[0]] = { description: '0%', range: (0..0), low: 0, high: 0 }
+        m_colors[colors[1]] = { description: 'Any - 10%', range: (0.000001..10.0), low: 0.000001, high: 10.0 }
+        m_colors[colors[2]] = { description: '11% - 15%', range: (10.000001..15.0), low: 10.000001, high: 15.0 }
+        m_colors[colors[3]] = { description: '15% - 20%', range: (15.000001..20.0), low: 15.000001, high: 20.0 }
+        m_colors[colors[4]] = { description: '20% - 25%', range: (20.000001..25.0), low: 20.000001, high: 25.0 }
+        m_colors[colors[5]] = { description: '25%+', range: (25.000001..100.0), low: 25.000001, high: 100.0 }
```

Comment:
Do these match up with the rounding functions?

Reply:
They should be granular enough to catch what we need.